scripts/check_backend_associated_test_file_test.py

```diff
@@ -44,27 +46,31 @@ def mock_error(msg: str) -> None:
         self.print_swap = self.swap(builtins, 'print', mock_print)
         self.swap_logging = self.swap(logging, 'error', mock_error)
         self.swap_exit = self.swap(sys, 'exit', lambda _: None)
+
+    def test_checks_fail_when_a_backend_file_lacks_associated_test_file(
+            self) -> None:
         # Creating a backend file.
-        with open(self.backend_file, 'w', encoding='utf8') as f:
-            f.write('Example code')
+        temp_file1 = tempfile.NamedTemporaryFile()
+        setattr(temp_file1, 'name', self.backend_file)
         # Creating a frontend file.
-        with open(self.frontend_file, 'w', encoding='utf8') as f:
+        temp_file2 = tempfile.NamedTemporaryFile()
+        setattr(temp_file2, 'name', self.frontend_file)
+        with utils.open_file(self.backend_file, 'w', encoding='utf8') as f:
+            f.write('Example code')
+        with utils.open_file(self.frontend_file, 'w', encoding='utf8') as f:
             f.write('Example code')
 
-    def tearDown(self) -> None:
-        super().tearDown()
-        os.remove(os.path.join(os.getcwd(), self.backend_file))
-        os.remove(os.path.join(os.getcwd(), self.frontend_file))
-        backend_test_file_path = os.path.join(
-            os.getcwd(), self.backend_test_file)
-        if os.path.exists(backend_test_file_path):
-            os.remove(backend_test_file_path)
-
-    def test_checks_fail_when_a_backend_file_lacks_associated_test_file(
-            self) -> None:
         with self.print_swap, self.swap_logging, self.swap_exit:
             check_backend_associated_test_file.main()
 
+        temp_file1.close()
+        temp_file2.close()
+        # On Windows system, occasionally this temp file is not deleted.
+        if os.path.isfile(self.backend_file):
+            os.remove(self.backend_file)
+        if os.path.isfile(self.frontend_file):
+            os.remove(self.frontend_file)
```

Comment:
I don't think you need this since we don't ever develop directly on windows (we use WSL2)

Reply:
Done